### ISSUE TITLE
Red exclamation mark on build.gradle files

### ISSUE THREAD

**Issue by mnlipp** [groovy/groovy-eclipse#266]
Since I have installed eclipse-groovy (neon snapshot), I have red exclamation marks on all gradle related files (build.gradle, settings.gradle). Things still work, but I'd like to have my project "problem free". Is there any way to get rid of the exclamation marks?

(I _did_ search the web. But I found nothing.)

**Comment by eric-milles** [comment]
The ! overlay is an indication that the file type is recognized as Groovy, but the Groovy nature is not present on your project.  There are at least 2 ways to go from here:
1) Remove .gradle from the Groovy source content type for your project or workspace
2) Add the Groovy nature to your project

**Comment by tobias-jenkner** [comment]
This does not work for me:
1. this does not work for me since the content types are locked in my eclipse (I cannot change the gradle/groovy association). 
2. I do not want to add the groovy nature only because I have a gradle build. 

With a previous version of eclipse-groovy the red exclamation marks did not appear for gradle files. I would like to have that as default behaviour again.

**Comment by eric-milles** [comment]
Am I understanding correctly that you have Groovy-Eclipse installed, but not Buildship (Gradle)?

I have been undergoing an effort to refresh the icons for Groovy-Eclipse.  To that end, I have been reducing the surface area by using base JDT icons in some situations.  The build.gradle file is one of those situatioins.  When a project is not configured with the Groovy nature and contains a file under the Groovy content type, an indicator is added to tell the user that the build path does not apply to files.  In earlier versions of Groovy-Eclipse, there was an icon for this that has been replaced with the exclamation mark decoration.

To sum up, the build path indication on build.gradle, et al. is not new.  It is just more clear using the current iconography.

**Comment by tobias-jenkner** [comment]
Exactly I have Groovy-Eclipse installed and not Buildship (I do not find Buildship very convincing at the moment). 

My projects just look like in the following screenshots since I have a lot of gradle files: 
![2017-04-19 17_21_51-smart_subversive2 - java - eclipse](https://cloud.githubusercontent.com/assets/1155047/25187934/c01cd890-2524-11e7-9f61-0997ff98c842.png)

If there are so many red exclamation marks this is a bit disturbing to me, since the red exclamation marks appear usually in a situation where there is really a build problem in my java projects.

**Comment by eric-milles** [comment]
If you add Groovy script filters (that is, non-compiled file patterns), the exclamation may not be displayed -- not sure if this meshes with your exclusion of the Groovy Nature.  This can be done by selecting Properties from the project's context menu, selecting Groovy Compiler from list, checking 'Enable project specific settings' and 'Enable script folder support'.  Then click Add and enter '**/*.gradle'.

**Comment by eric-milles** [comment]
Here is what it looks like in my project:
![script-filters](https://cloud.githubusercontent.com/assets/18193802/25193036/2fd8b1e2-24fb-11e7-9f7a-d5e3c4627df9.png)

**Comment by eric-milles** [comment]
You can also add the Groovy Nature without the Java Builder.  You mention that Gradle handles the builds.

**Comment by tobias-jenkner** [comment]
Thanks a lot for your comments. This works for me, but nevertheless I think that the default behaviour of having these big red exclamation marks in projects that are not groovy projects on gradle files is not very good. If I added the groovy nature then it would make sense to me.

**Comment by eric-milles** [comment]
I have a plan to set *.gradle as a script exclusion by default.  I'm trying to pick away at the problems a little at a time.  There is a lot of stuff that was setup before my time.

**Comment by tobias-jenkner** [comment]
Excellent. Thank you very much for your work! I am really happy that you provide groovy tooling for eclipse. When I criticize some points (like above), I just want to improve your product. 
Again: Thank you very much!
